src/cast_core/include/cast_core/actors/MonotonicSingleLoader.hpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_A0EBCED8_5FE8_4EA0_9292_52531417B62E_INCLUDED
+#define HEADER_A0EBCED8_5FE8_4EA0_9292_52531417B62E_INCLUDED
+
+#include <string_view>
+
+#include <mongocxx/pool.hpp>
+
+#include <gennylib/Actor.hpp>
+#include <gennylib/PhaseLoop.hpp>
+#include <gennylib/context.hpp>
+
+#include <metrics/metrics.hpp>
+
+namespace genny::actor {
+
+/**
+ * Bulk inserts data into a single collection using multiple threads. Similar to
+ * the MonotonicLoader actor, the generated documents have a monotonically
+ * increasing _id, starting from {_id: 0}.
+ *
+ * However, the MonotonicSingleLoader actor differs from the Loader and
+ * MonotonicLoader actors in a few notable ways:
+ *
+ *  - The collection name is optional and defaults to "Collection0" if omitted.
+ *
+ *  - The MonotonicSingleLoader actor must only ever be active in one phase of the
```

Comment:
Should we also mention that "MonotonicSingleLoader" may only load data into a single collection?